widget/image.go

```diff
@@ -53,3 +60,143 @@ func (im Image) Layout(gtx layout.Context) layout.Dimensions {
 
 	return dims
 }
+
+type ImageSequences struct {
+	// Fit specifies how to scale the image to the constraints.
+	// By default it does not do any scaling.
+	Fit Fit
+	// Position specifies where to position the image within
+	// the constraints.
+	Position layout.Direction
+	// Frames is the sequence images, each image must be independent
+	// from the previous one and only one will be drawed per frame
+	Frames []paint.ImageOp
+	// FramesEnds control when each frame ends. The FramesEnds must
+	// have the lenght of Frames.
+	FramesEnds []time.Duration
```

Comment:
The `image/gif` uses `Delay` for each frame. I choose to create some kind of `FramesEnds` to make easier to skip frames, in case Gio can't create frames as fast as the delay.

However, I think it could have a "fast path" for constant delays. Consider something like `ConstantDelay`. That will remove the slice and each "End" could be calculated by  `frame * delay`. Most GIFs uses the same "Delay" for all frame, in that case the slice seems overkill.